messybrainz/fetch_artist_mbids.py

```diff
@@ -0,0 +1,117 @@
+# Script to fetch artist MBIDs from MusicBrainz Database using
+# the recording MBIDs in the MessyBrainz database.
+
+import brainzutils.musicbrainz_db.recording as mb_recording
+import json
+import logging
+import re
+
+import messybrainz.default_config as config
+try:
+    import messybrainz.custom_config as config
+except ImportError:
+    pass
+
+from brainzutils import cache
+from brainzutils import musicbrainz_db
+from brainzutils.musicbrainz_db.exceptions import NoDataFoundException
+from messybrainz import db
+from sqlalchemy import text
+
+
+def check_valid_uuid(s):
+    """ Checks if the recording MBID is in the valid format or not.
+        Returns True if its a valid UUID else returns False.
+    """
+    if re.match(r'^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$', s):
+        return True
+    else:
+        return False
+
+
+def add_artist_mbids(recording_mbid, artist_mbids):
+    """ Inserts the artist_mbids corresponding to the recording_mbids
+        into the recording_artist table.
+    """
+    with db.engine.begin() as connection:
+        for artist_mbid in artist_mbids:
+            query = text("""INSERT INTO recording_artist (recording_mbid, artist_mbid)
+                            VALUES (:recording_mbid, :artist_mbid)
+                            RETURNING recording_mbid""")
+            result = connection.execute(query, {
+            "recording_mbid": recording_mbid,
+            "artist_mbid": artist_mbid,
+            })
+
+    return is_recording_mbid_present(recording_mbid)
+
+
+def is_recording_mbid_present(recording_mbid):
+    """
+        Check if recording MBID is already present in recording_artist table.
+        Returns True if recording MBID is present else False is returned.
+    """
+    with db.engine.begin() as connection:
+        query = text("""SELECT recording_mbid FROM recording_artist
+                        WHERE recording_mbid = :recording_mbid""")
+        result = connection.execute(query, {
+        "recording_mbid": recording_mbid,
+        })
+
+        if result.rowcount:
+            return True
+
+    return False
+
+
+def fetch_artist_mbids(recording_mbid):
+    """ Fetches artist MBIDs from the MusicBrainz database for the recording MBID.
+        And inserts the artist MBIDs into the recording_artist table.
+    """
+    try :
+        recording = mb_recording.get_recording_by_mbid(recording_mbid, include=['artists'])
+    except NoDataFoundException:
+        return False
+
+    artist_mbids = []
+    for artist in recording['artists']:
+        artist_mbids.append(artist['id'])
+
+    result = add_artist_mbids(recording_mbid, artist_mbids)
```

Comment:
You have a function called fetch_artist_mbids() -- from a caller's perspective, this function should not alter the database. If this is how you want the function to really work, you should change the name to something that indicates that the database state my be changed. fetch_and_insert_artist_mbids() could be a better name... But, more likely you should examine your algorithmic flow and see if this is perhaps better broken into two separate functions...